tensorflow_lite_support/cc/task/processor/image_postprocessor.cc

```diff
@@ -0,0 +1,181 @@
+namespace processor {
+
+/* static */
+tflite::support::StatusOr<std::unique_ptr<ImagePostprocessor>>
+ImagePostprocessor::Create(core::TfLiteEngine* engine,
+                           const std::initializer_list<int> output_indices,
+                           const std::initializer_list<int> input_indices) {
+  RETURN_IF_ERROR(Postprocessor::SanityCheck(/* num_expected_tensors = */ 1,
+                                             engine, output_indices));
+  auto processor =
+      absl::WrapUnique(new ImagePostprocessor(engine, output_indices));
+
+  RETURN_IF_ERROR(processor->Init(input_indices));
+  return processor;
+}
+
+absl::Status ImagePostprocessor::Init(const std::vector<int>& input_indices) {
+  if (Tensor()->type != kTfLiteUInt8 && Tensor()->type != kTfLiteFloat32) {
+    return tflite::support::CreateStatusWithPayload(
+        absl::StatusCode::kInvalidArgument,
+        absl::StrFormat("Type mismatch for output tensor %s. Requested one "
+                        "of these types: "
+                        "kTfLiteUint8/kTfLiteFloat32, got %s.",
+                        Tensor()->name, TfLiteTypeGetName(Tensor()->type)),
+        tflite::support::TfLiteSupportStatus::kInvalidOutputTensorTypeError);
+  }
+
+  // Check output tensor dimensions.
+  if (Tensor()->dims->size != 4) {
+    return tflite::support::CreateStatusWithPayload(
+        absl::StatusCode::kInvalidArgument,
+        absl::StrFormat(
+            "Output tensor is expected to have 4 dimensions, found %d.",
+            Tensor()->dims->size),
+        tflite::support::TfLiteSupportStatus::
+            kInvalidOutputTensorDimensionsError);
+  }
+
+  if (Tensor()->dims->data[0] != 1) {
+    return tflite::support::CreateStatusWithPayload(
+        absl::StatusCode::kInvalidArgument,
+        absl::StrFormat("Expected batch size of 1, found %d.",
+                        Tensor()->dims->data[0]),
+        tflite::support::TfLiteSupportStatus::
+            kInvalidOutputTensorDimensionsError);
+  }
+
+  // RGB check.
+  if (Tensor()->dims->data[3] != 3) {
+    return tflite::support::CreateStatusWithPayload(
+        absl::StatusCode::kInvalidArgument,
+        absl::StrFormat("Expected depth size of 3, found %d.",
+                        Tensor()->dims->data[3]),
+        tflite::support::TfLiteSupportStatus::
+            kInvalidOutputTensorDimensionsError);
+  }
+
+  // Gather metadata
+  auto* output_metadata =
+      engine_->metadata_extractor()->GetOutputTensorMetadata(
+          output_indices_.at(0));
+  auto* input_metadata = engine_->metadata_extractor()->GetInputTensorMetadata(
+      input_indices.at(0));
+
+  // Use input metadata for normalization as fallback.
+  auto* processing_metadata =
+      output_metadata != nullptr ? output_metadata : input_metadata;
+
+  absl::optional<vision::NormalizationOptions> normalization_options;
+  ASSIGN_OR_RETURN(normalization_options,
+                   GetNormalizationOptionsIfAny(*processing_metadata));
```

Comment:
Since this was wrapped inside `unknown namespace` inside `image_tensor_specs.cc` we might need to copy-paste the code here unfortunately.

Reply:
Please share as much code as possible between`ImagePostprocessor::Init` and `ImageTensorSpecs::BuildInputImageTensorSpecs`. You can put a todo and implement it in a follow up PR.

Reply:
Just so we're clear, do you mean copy code from `BuildInputImageTensorSpecs` when you said "share".

Reply:
"Share code" means `ImagePostprocessor::Init` and `ImageTensorSpecs::BuildInputImageTensorSpecs` use the same piece of code to do processing or validation.